### PR TITLE
Added web eslint config

### PR DESCRIPTION
**Pull request by sameerk129**
[Asana Ticket](https://app.asana.com/0/1202395086141153/1205177885438027/f)

### Summary
Maintaining eslint web configurations in `web/` folder.
Updated the version in package.json
Added .idea/ to .gitignore

### CODE REVIEW

File: src/config/web/recommended-typescript-rules.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  ],
+  'import/no-extraneous-dependencies': [
+    'error',
+    {
+      devDependencies: [
+        '**/stories.*',
+        '**/.storybook/**/*.*',
+        'apps/**/partner_react/**/*.test.ts',
+        'apps/**/partner_react/**/*.test.tsx',
+        'apps/**/partner_react/**/test/**',
```
Comment (ashwinibm) [inline src/config/web/recommended-typescript-rules.js:87]: Do we still have partner_react/ folders?

File: src/config/web/recommended-typescript-rules.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        'web_frontend/**/*.test.ts',
+        'web_frontend/**/*.test.tsx',
+        'web_frontend/**/test/**',
+        'web_frontend/**/__mocks__/*.ts',
+        'web_frontend/common/src/test/setup-tests.ts',
```
Comment (ashwinibm) [inline src/config/web/recommended-typescript-rules.js:92]: Could you check if we can avoid the instawork repo specific prefixes? For e.g. the mocha tests have `*.spec.ts` pattern so it should be ok to just use `*.test.tsx` pattern here.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-plugin-instawork",
-  "version": "0.11.0",
+  "version": "0.12.0",
```
Comment (flochtililoch) [inline package.json:3]: this should be done via `yarn release:minor` from master branch

```suggestion
  "version": "0.11.0",
```

File: package.json [hunk 4]
```diff
@@ -30,6 +30,7 @@
     "eslint-config-airbnb": "18.2.1",
     "eslint-config-prettier": "8.0.0",
     "eslint-import-resolver-babel-module": "5.2.0",
+    "eslint-import-resolver-typescript": "^3.6.0",
```
Comment (flochtililoch) [inline package.json:33]: let's pin this (you'll need to run `yarn` again before merging the PR)
```suggestion
    "eslint-import-resolver-typescript": "3.6.0",
```

Comment (ashwinibm) [reply package.json]: Fixed this.

File: package.json (hunk 3 above)
Comment (ashwinibm) [reply package.json]: Fixed it. Will run the release from master once this is merged.

**Review by flochtililoch — APPROVED** [review]

**Pull request merged**